metrics/latency.go

```diff
@@ -73,6 +73,12 @@ func ProbeLatency(ch channel2.Channel, histogram Histogram, interval time.Durati
 			}
 		case <-time.After(timeout):
 			log.Errorf("latency timeout after [%s]", timeout)
+			if ch.GetTimeSinceLastRead() > interval {
+				// No traffic on channel, no response. Close the channel
+				log.Error("no read traffic on channel since before latency probe was sent, closing channel")
+				_ = ch.Close()
```

Comment:
Not sure how I feel about this. At the very least we probably want this to be disabled by default, and only enabled in certain cases (assuming this is related to the interface switch issue in the edge?)

Maybe we want some kind of `TimeoutHandler` that can make a decision what to do with this kind of timeout, rather than having the channel close itself?

Reply:
Right now it's used in two place, sdk <-> router and router <-> router links. I could see wanting this in both places, but agree there should be a way to be explicit about if you want it. I'll see how the timeout handler looks

Reply:
Added ResultHandler,TimeoutHandler and ExitHandler